### PR TITLE
added BSN

### PR DESCRIPTION
**Pull request by abbert**
```markdown
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -
```

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,3 +3,4 @@
 /build/*
 /bin/*
 /phpunit.xml
+.idea
```
Comment (Soullivaneuh) [inline .gitignore]: This have nothing to be here. This is specific to your dev env.

https://help.github.com/articles/ignoring-files/#create-a-global-gitignore

File: src/IsoCodes/Bsn.php [hunk 2]
```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace IsoCodes;
+
+/**
+ * Class Bsn.
```
Comment (Soullivaneuh) [inline src/IsoCodes/Bsn.php]: This comment is not relevant, please delete it.

File: src/IsoCodes/Bsn.php [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+ *
+ * In the Netherlands, the citizen service number (BSN) is a unique personal number
+ * allocated to everyone registered in the Municipal Personal Records Database
+ */
+class Bsn implements IsoCodeInterface
```
Comment (Soullivaneuh) [inline src/IsoCodes/Bsn.php]: I think this class should not be extended and should be `final`. cc @ronanguilloux

File: tests/IsoCodes/Tests/BsnTest.php [hunk 4]
```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace IsoCodes\Tests;
+
+/**
+ * @covers Isocodes\Bsn
+ *
+ * @author Albert Bakker <hello@abbert.nl>
+ */
+class BsnTest extends AbstractIsoCodeInterfaceTest
```
Comment (Soullivaneuh) [inline tests/IsoCodes/Tests/BsnTest.php]: `final`. https://github.com/ronanguilloux/IsoCodes/pull/97/files#r63833959

File: tests/IsoCodes/Tests/BsnTest.php [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+        ];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getInvalidValues()
+    {
+        return [
+            ['2018453'],     // < 8 chars
```
Comment (Soullivaneuh) [inline tests/IsoCodes/Tests/BsnTest.php]: Could you please indent your comments?

**Comment by Soullivaneuh** [comment]
Could you add a `@link` like here? https://github.com/ronanguilloux/IsoCodes/blob/a31dd77bd4b67674e028ca6100e972aff3fbdd2f/src/IsoCodes/Cif.php#L14

I would like an english documentation, but I only found this one: https://nl.wikipedia.org/wiki/Burgerservicenummer

**Comment by abbert** [comment]
Added .idea folder :scream: Deleted it now

**Comment by abbert** [comment]
This is not going to work

**Pull request closed without merging**